spec/safe_statements_spec.rb

```diff
@@ -1343,6 +1343,23 @@ def up
           end
         end
 
+        describe "safe_make_column_not_nullable" do
+          it "adds the not null constraint to the column" do
+            migration = Class.new(migration_klass) do
+              def up
+                unsafe_create_table :foos do |t|
+                  t.text :bar, :null => false
+                end
+                safe_make_column_not_nullable :foos, :bar
+              end
+            end
+
+            migration.suppress_messages { migration.migrate(:up) }
```

Comment:
I wonder if we should add an assertion that this doesn't use locks beyond a certain level.

@rkrage Thoughts on this? I don't think we have this, but I'd love to have a test helper something like this:

```
expect do
  ...
end.not_to acquire_locks_on_table(...).higher_than(...)
```

Or similar.

Thinking out loud here:

I don't think there's a way to get Postgres to log all locks acquired, and so we'd probably have to do something like lock the table and then run the second query to verify what type is waiting in `pg_locks`. That doesn't work easily for multiple statements, though, so I think in this case we might lock the table at a lower level and expect this to proceed. But I think the `ALTER TABLE` will still need an AccessExclusiveLock, it will just be very short lived.

Reply:
I'll have to think about this a bit, but I think it's going to be super difficult to make a generic helper. As you pointed out, we could acquire an `AccessExclusiveLock` and sleep for x seconds in a thread and then concurrently run the code under test. That code would block until the sleep expires and then we could go inspect the logs for the output from `log_lock_waits`. However, that breaks down for methods that execute multiple SQL statements.

I guess we could do something crazy and mock the `connection#execute` method with `and_wrap_original` such that it automatically does the AEL shenanigans described above before _any_ SQL is executed

Reply:
Since we don't currently have a precedent for writing such a test, I'm not sure if it's fair to put this on @alexspeller 😄 

This can maybe be something we do as a follow on to this PR

Reply:
Yeah realistically I don't have the bandwidth at the moment to deep dive on that unfortunately - I didn't see any precedent for more in-depth lock testing for similar methods so I kept it quite basic

Reply:
Yeah, I think we're fine for now; I wanted to at least kick off the conversation internally if nothing else.